src/rust/ensogl/src/display/scene.rs

```diff
@@ -151,6 +191,57 @@ impl Default for Target {
 }
 
 
+// === Target Tests ===
+
+#[cfg(test)]
+mod target_tests {
+    use super::*;
+
+    fn assert_valid_roundtrip(value1:u32, value2:u32) {
+        let pack   = Target::encode(value1,value2);
+        let unpack = Target::decode(pack.0.into(),pack.1.into(),pack.2.into());
+        assert_eq!(unpack.0,value1);
+        assert_eq!(unpack.1,value2);
+    }
```

Comment:
It would be cool to have docs here. I don't understand why you want this to compile now: `assert_valid_roundtrip(1024, 999)`